src/rpm/builder.rs

```diff
@@ -64,12 +106,37 @@ pub struct PackageBuilder {
     enhances: Vec<Dependency>,
     supplements: Vec<Dependency>,
 
-    pre_trans_script: Option<String>,
     pre_inst_script: Option<String>,
```

Comment:
It might make sense to treat these like `Dependency` and create a common struct.

That is:

```
struct Scriptlet {
    script: Option<String>,
    flags: Option<u32>,
    prog: Option<Vec<String>>,
}
```

Reply:
@dralley Cool, but I'll leave the original builder methods for back-compat. Should I add a [deprecate] tag though?

Reply:
There's no particular need to keep them for compat.  It'll just lead to more churn if we use a worse name and have to rename them back later.  Plus it ought to be a simple fix for the limited cases that were previously supported.

Reply:
@dralley I didn't happen to see this message before I pushed this most recent iteration, but let me know what you think. I think it's a reasonable middle-ground since setting flags/program is optional I optimized for setting scriptlets w/o the added config in the builder. For more advanced scenarios, I added a `scriptlet` method to the package builder.